ipatests/test_integration/test_insecure_crypto_SCEP.py

```diff
@@ -0,0 +1,141 @@
+#
+# Copyright (C) 2019  FreeIPA Contributors see COPYING for license
+#
+
+import glob
+
+from ipaplatform.paths import paths
+
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+AUTHORIZATION_FILE = '/var/lib/pki/pki-tomcat/ca/conf/flatfile.txt'
+CERTMONGER_CONFIG = '/etc/sysconfig/certmonger'
+
+
+def enable_SCEP(host):
+    """Enable SCEP in Certificate Authority's CS.cfg
+    by setting ca.scep.enable=true
+    """
+    content = host.get_file_contents(paths.CA_CS_CFG_PATH,
+                                     encoding='utf-8')
+    new_lines = []
+    input_line = "auths.instance.flatFileAuth.deferOnFailure=false"
+    for line in content.split('\n'):
+        if line.startswith('auths.instance.flatFileAuth.deferOnFailure'):
+            new_lines.append(input_line)
+        elif line.startswith('ca.scep.enable'):
+            new_lines.append("ca.scep.enable=true")
+        else:
+            new_lines.append(line)
+    host.put_file_contents(paths.CA_CS_CFG_PATH, '\n'.join(new_lines))
+
+
+def add_local_ip_to_authfile(host):
+    """Add the local IP to the authorization file,
+    /var/lib/pki/pki-tomcat/ca/conf/flatfile.txt
+    """
+    content = host.get_file_contents(AUTHORIZATION_FILE,
+                                     encoding='utf-8')
+    new_lines = []
+    for line in content.split('\n'):
+        if line.startswith('#UID:'):
+            new_lines.append("UID:%s" % host.ip)
+        elif line.startswith("#PWD:"):
+            new_lines.append("PWD:1234")
+        else:
+            new_lines.append(line)
+    host.put_file_contents(AUTHORIZATION_FILE, '\n'.join(new_lines))
+
+
+def edit_certmonger_config(host):
+    """Increase certmonger debug output by setting
+    OPTS=-d 3 in /etc/sysconfig/certmonger
+    """
+    content = host.get_file_contents(CERTMONGER_CONFIG,
+                                     encoding='utf-8')
+    new_lines = []
+    for line in content.split('\n'):
+        if line.startswith('OPTS='):
+            new_lines.append("OPTS=-d 3")
+        else:
+            new_lines.append(line)
+    host.put_file_contents(CERTMONGER_CONFIG, '\n'.join(new_lines))
+
+
+def set_scep_cipher(host):
+    """hard code the default SCEP cipher to DES3 since there
+    is no CLI to do this by adding scep_cipher=DES3
+    """
+    input_file = None
+    for file in glob.glob('/var/lib/certmonger/cas/*'):
```

Comment:
``file`` was a reserved name in Python 2. We still avoid the name. Please use ``filename`` instead.

The function assumes that the controller machine has certmonger installed. This may not be the case. Please use ``host.run_command(['find', '/var/lib/certmonger/cas', '-type', 'f'])`` to get a list of file names, then retrieve the file from the host.